modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/snapshot/CurrencySnapshotCreator.scala

```diff
@@ -132,6 +135,7 @@ object CurrencySnapshotCreator {
 
           feeTransactions = dataApplicationAcceptanceResult
             .map(result => SortedSet.from(result.feeTransactions))
+            .orElse(maybeFeeTransactions)
```

Comment:
I think the fallback should be there only if there is no data application implementation provided

Reply:
@marcinwadon Isn't that what the above is doing?

- `dataApplicationAcceptanceResult` is nonEmpty if `dataApplicationSnapshotAcceptanceManager` is nonEmpty
- `dataApplicationSnapshotAcceptanceManager` is nonEmpty if the data application (and calculated state storage) are nonEmpty

Reply:
Ok. I didn't check the inside implementation of `dataApplicationAcceptanceResult`. If it's non empty if and only if data application is implemented, then it's fine :+1: Looking for this part only, it looks for me that the result may be empty if there is any internal error of the data application acceptance (and we should throw then, not fallback). but if that's not the case then it's okay :)